index.js

```diff
@@ -29,11 +29,22 @@ function postHTMLParser(html, options) {
         var directives = options.directives || defaultDirectives;
         var last = bufArray.last();
 
+        var tagName;
         for (var i = 0; i < directives.length; i++) {
             var directive = directives[i];
             var directiveText = directive.start + data + directive.end;
+            var isDirective = false;
```

Comment:
```js
function isDirective (directive, tag) {
  if (directive.name instanceof RegExp) {
    const regex = new RegExp(directive.name, 'i')

    return regex.test(tag)
  }

  if (tag !== directive.name) {
    return false
  }
  
  return true
}

Reply:
Untested but please a make this a private helper function instead of `n` variables and `{Boolean}` switches/assignments

Reply:
this line ```const regex = RegExp(directive.name, 'i')```
```const``` is not supported by es5 and ```RegExp(directive.name, 'i')``` only works in ES6.
Related to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp)
> Starting with ECMAScript 6, new RegExp(/ab+c/, 'i') no longer throws a TypeError ("can't supply flags when constructing one RegExp from another") when the first argument is a RegExp and the second flags argument is present. A new RegExp from the arguments is created instead.

So we change the ```const``` to ```var``` and ```RegExp(directive.name, 'i')``` to ```RegExp(directive.name.source, 'i')```
Thank you

Reply:
```
var pattern = /a/

var regex = RegExp(pattern, 'i')

console.log(regex.test('Abc'))
```

⚠️ Note that it's `RegExp(pattern, flags)` **not** `new RegExp(string, flags)` (No `new`)